File: Data Structures & Algorithms/two-integer-sum/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        const numsMap = new Map();

    for (let i = 0; i < nums.length; i++) {
        const diff = target - nums[i];
        if (numsMap.has(diff)) {
            return [numsMap.get(diff), i]; // correct order: earlier index first
        }
        numsMap.set(nums[i], i);
    }
    }
}

File: Data Structures & Algorithms/binary-search/submission-1.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        
        left, right = 0, len(nums) - 1
        mid = int((left + right) / 2)

        while left <= right:
            if target == nums[mid]:
                return mid
            elif target < nums[mid]:
                right = mid - 1
                mid = int((left + right) / 2)
            else:
                left = mid + 1
                mid = int((left + right) / 2)

        return -1